port/linux/omrvmem.c

```diff
@@ -1075,62 +1092,94 @@ get_hugepages_info(struct OMRPortLibrary *portLibrary, vmem_hugepage_info_t *pag
-	int ft = -1;
 	int flags = 0;
 	void *result = NULL;
 	int protectionFlags = PROT_NONE;
-	BOOLEAN useBackingSharedFile = FALSE;
 	BOOLEAN useBackingFile = FALSE;
+	BOOLEAN useBackingSharedFile = FALSE;
+	BOOLEAN areHugePagesEnabled = FALSE;
 
-	Trc_PRT_vmem_default_reserve_entry(address, byteAmount);
+	Trc_PRT_vmem_reserve_entry(address, byteAmount);
 
-	if(mode & OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN) {
+	if (0 != (mode & OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN)) {
 		flags |= MAP_SHARED;
 		useBackingSharedFile = TRUE;
 	} else {
-#if defined(MAP_ANONYMOUS)
-		flags |= MAP_ANONYMOUS | MAP_PRIVATE;
-		useBackingSharedFile = FALSE;
-#elif defined(MAP_ANON)
-		flags |= MAP_ANON | MAP_PRIVATE;
-		useBackingSharedFile = FALSE;
-#else
-		useBackingFile = TRUE;
-		flags |= MAP_PRIVATE;
-#endif
+		useBackingFile = set_flags_for_mmap(&flags);
+	}
+
+	if (0 != (mode & OMRPORT_VMEM_MEMORY_MODE_MMAP_HUGE_PAGES)) {
+		flags |= MAP_HUGETLB;
+		areHugePagesEnabled = TRUE;
 	}
 
 	if (useBackingSharedFile) {
-		char filename[FILE_NAME_SIZE + 1];
-		sprintf(filename, "omrvmem_temp_%zu_%09d_%zu",  (size_t)omrtime_current_time_millis(portLibrary),
+		memfd_call memfd_createDL = (memfd_call)PPG_memfd_call;
+		if (NULL != memfd_createDL) {
+			Trc_PRT_vmem_reserve_doubleMapAPI_available(address, byteAmount);
+			mode |= OMRPORT_VMEM_MEMORY_MODE_DOUBLE_MAP_AVAILABLE;
+			int ft = -1;
+			uintptr_t fdFlags = 0;
+			char filename[FILE_NAME_SIZE + 1];
+			sprintf(filename, "omrvmem_temp_%zu_%09d_%zu",  (size_t)omrtime_current_time_millis(portLibrary),
 								getpid(),
 								(size_t)pthread_self());
-		fd = shm_open(filename, O_RDWR | O_CREAT, 0600);
-		shm_unlink(filename);
-		ft = ftruncate(fd, byteAmount);
-
-		if (fd == -1 || ft == -1) {
-			Trc_PRT_vmem_default_reserve_failed(address, byteAmount);
-			update_vmemIdentifier(identifier, NULL, NULL, 0, 0, 0, 0, 0, NULL, -1);
-			Trc_PRT_vmem_default_reserve_exit(result, address, byteAmount);
-			return result;
+
+			if (areHugePagesEnabled) {
+				fdFlags = MFD_HUGETLB;
+			}
+			fd = memfd_createDL(filename, fdFlags);
+			ft = ftruncate(fd, byteAmount);
+
+			if ((-1 == fd) || (-1 == ft)) {
+				Trc_PRT_vmem_reserve_failed(address, byteAmount);
+				update_vmemIdentifier(identifier, NULL, NULL, 0, 0, 0, 0, 0, NULL, -1);
+				Trc_PRT_vmem_reserve_exit(result, address, byteAmount);
+				return result;
+			}
+		} else {
+			flags = 0;
+			useBackingFile = set_flags_for_mmap(&flags);
+			useBackingSharedFile = FALSE;
+			Trc_PRT_vmem_reserve_doubleMapAPI_not_available(address, byteAmount);
 		}
-	} else if(useBackingFile) {
+	} else if (useBackingFile) {
 		fd = portLibrary->file_open(portLibrary, "/dev/zero", EsOpenRead | EsOpenWrite, 0);
 	}
 
-	if((fd == -1 && (!useBackingSharedFile && !useBackingFile)) ||
-	   (fd != -1 && (useBackingSharedFile || useBackingFile))) {
+	if (((-1 == fd) && (!useBackingSharedFile && !useBackingFile)) ||
+	   ((-1 != fd) && (useBackingSharedFile || useBackingFile))) {
```

Comment:
There are a couple of places you can use `OMRPORT_INVALID_FD` instead of `-1`
```suggestion
	if (((OMRPORT_INVALID_FD == fd) && (!useBackingSharedFile && !useBackingFile)) ||
	   ((OMRPORT_INVALID_FD != fd) && (useBackingSharedFile || useBackingFile))) {
```